kafka-junit-core/src/main/java/com/salesforce/kafka/test/KafkaTestServer.java

```diff
@@ -129,46 +155,55 @@ public void start() throws Exception {
         zkServer = new TestingServer(zkInstanceSpec, true);
         final String zkConnectionString = getZookeeperServer().getConnectString();
 
-        // Create temp path to store logs
-        final File logDir = Files.createTempDir();
-        logDir.deleteOnExit();
+        // Build properties using a baseline from overrideBrokerProperties.
+        final Properties brokerProperties = new Properties(overrideBrokerProperties);
+
+        // Put required zookeeper connection properties.
+        setPropertyIfNotSet(brokerProperties, "zookeeper.connect", zkConnectionString);
+
+        // Conditionally generate a port for kafka to use if not already defined.
+        kafkaPort = (String) setPropertyIfNotSet(brokerProperties, "port", String.valueOf(InstanceSpec.getRandomPort()));
 
-        // Determine what port to run kafka on
-        kafkaPort = String.valueOf(InstanceSpec.getRandomPort());
+        // If log.dir is not set.
+        if (brokerProperties.getProperty("log.dir") == null) {
+            // Create temp path to store logs
+            final File logDir = Files.createTempDir();
+            logDir.deleteOnExit();
 
-        // Build properties
-        Properties kafkaProperties = new Properties();
-        kafkaProperties.setProperty("zookeeper.connect", zkConnectionString);
-        kafkaProperties.setProperty("port", kafkaPort);
-        kafkaProperties.setProperty("log.dir", logDir.getAbsolutePath());
-        kafkaProperties.setProperty("auto.create.topics.enable", "true");
-        kafkaProperties.setProperty("zookeeper.session.timeout.ms", "30000");
-        kafkaProperties.setProperty("broker.id", "1");
-        kafkaProperties.setProperty("auto.offset.reset", "latest");
+            // Set property.
+            brokerProperties.setProperty("log.dir", logDir.getAbsolutePath());
+        }
 
         // Ensure that we're advertising appropriately
-        kafkaProperties.setProperty("host.name", localHostname);
-        kafkaProperties.setProperty("advertised.host.name", localHostname);
-        kafkaProperties.setProperty("advertised.port", kafkaPort);
-        kafkaProperties.setProperty("advertised.listeners", "PLAINTEXT://" + localHostname + ":" + kafkaPort);
-        kafkaProperties.setProperty("listeners", "PLAINTEXT://" + localHostname + ":" + kafkaPort);
+        setPropertyIfNotSet(brokerProperties, "host.name", localHostname);
+        setPropertyIfNotSet(brokerProperties, "advertised.host.name", localHostname);
```

Comment:
I wonder if you should get the hostname property you set one lime above for this.

Reply:
done!